frappe_s3_attachment/controller.py

```diff
@@ -292,6 +305,91 @@ def upload_existing_files_s3(name, file_name):
     else:
         pass
 
+# download s3 files
+def download_s3_files(private_local_folder_path, public_local_folder_path):
+    s3_download = S3Operations()
+    bucket = s3_download.S3_RESOURCE.Bucket(s3_download.BUCKET)
```

Comment:
You are fetching the list of files from wrong place. There should always be a single source of truth, which in this case is the Files doctype. See how the upload function uses it to determine which files to upload. You must do the exact opposite to ensure it works.

In this case, it will break if I have additional files on my S3 bucket that don't exist in Files doctype.

Reply:
It will not break. I have handled that in my code. If any files which entries are not in database will not be downloaded.
For Uploading file they uses file list. But after File uploaded in s3, they modified the file name, attach underscore instead of spaces and attach hash as prefix of file.
For Ex, Tom 22-13.png -> PB7CEXG7_Tom_22-13.png